enterprise/server/invocation_stat_service/invocation_stat_service.go

```diff
@@ -112,7 +104,27 @@ func (i *InvocationStatService) GetTrend(ctx context.Context, req *inpb.GetTrend
 		q.AddWhereClause(fmt.Sprintf("(%s)", roleQuery), roleArgs...)
 	}
 
-	q.AddWhereClause(`updated_at_usec > ?`, timeutil.ToUsec(time.Now().Add(-lookbackWindowDays)))
+	if start := req.GetQuery().GetUpdatedAfter(); start.IsValid() {
+		q.AddWhereClause("updated_at_usec >= ?", timeutil.ToUsec(start.AsTime()))
+	} else {
+		// If no start time specified, respect the lookback window field if set,
+		// or default to 7 days.
+		lookbackWindowDays := 7 * 24 * time.Hour
+		if w := req.GetLookbackWindowDays(); w != 0 {
+			if w < 1 || w > 365 {
+				return nil, status.InvalidArgumentErrorf("lookback_window_days must be between 0 and 366")
+			}
+			lookbackWindowDays = time.Duration(w*24) * time.Hour
+		}
+		q.AddWhereClause("updated_at_usec >= ?", timeutil.ToUsec(time.Now().Add(-lookbackWindowDays)))
+	}
```

Comment:
Are there other places in the code we do this? if not, you can add a comment that this can be deleted in the next release (same for the proto field)

Reply:
Nope, we don't have other `updated_at` filters that would need to be migrated to the "new way." Added a comment here and to the proto as well.